osu.Game/OsuGame.cs

```diff
@@ -898,8 +899,44 @@ private void handleStartupImport()
             if (args?.Length > 0)
             {
                 string[] paths = args.Where(a => !a.StartsWith('-')).ToArray();
+
                 if (paths.Length > 0)
-                    Task.Run(() => Import(paths));
+                {
+                    string firstPath = paths.First();
+
+                    if (firstPath.StartsWith(OSU_PROTOCOL, StringComparison.Ordinal))
+                    {
+                        handleOsuProtocolUrl(firstPath);
+                    }
+                    else
+                    {
+                        Task.Run(() => Import(paths));
+                    }
+                }
+            }
+        }
+
+        private void handleOsuProtocolUrl(string url)
```

Comment:
Any reason to not use the existing `HandleLink(url)` method in `OsuGame`? That'll lead to the existing logic here:

https://github.com/ppy/osu/blob/d0d0830888159fcd92a110d84d2a27b5e8892680/osu.Game/Online/Chat/MessageFormatter.cs#L204-L243

Reply:
That whole structure `Link` / `LinkDetails` [was previously called out as pretty dodgy](https://github.com/ppy/osu/issues/11754) (and I agree), so maybe best not to? Not sure.

Reply:
I had a feeling this was already done somewhere, but couldn't immediately find the flow. I've switched to this as it seems pretty pointless to reimplement what we already have.